components/eamxx/src/physics/shoc/eamxx_shoc_process_interface.cpp

```diff
@@ -461,5 +465,44 @@ void SHOCMacrophysics::finalize_impl()
   // Do nothing
 }
 // =========================================================================================
+void SHOCMacrophysics::check_flux_state_consistency(const double dt)
+{
+  using PC = scream::physics::Constants<Real>;
+  const Real gravit = PC::gravit;
+
+  const auto& pseudo_density = get_field_in ("pseudo_density").get_view<const Spack**>();
+  const auto& surf_evap      = get_field_out("surf_evap").get_view<Real*>();
+  const auto& qv             = get_field_out("qv").get_view<Spack**>();
+
+  const auto nlevs      = m_num_levs;
+  const auto nlev_packs = ekat::npack<Spack>(nlevs);
+  const auto policy     = ekat::ExeSpaceUtils<KT::ExeSpace>::get_default_team_policy(m_num_cols, nlev_packs);
+  Kokkos::parallel_for("check_flux_state_consistency",
+                       policy,
+                       KOKKOS_LAMBDA (const KT::MemberType& team) {
+    const auto i = team.league_rank();
+
+    const auto& pseudo_density_i = ekat::subview(pseudo_density, i);
+    const auto& qv_i             = ekat::subview(qv, i);
 
+    if (surf_evap(i) < 0) {
+      const auto cc = abs(surf_evap(i)*dt*gravit);
+
+      auto qv_x_pdel = [&](const int k) {
+        return qv_i(k)*pseudo_density_i(k);
+      };
+      Real mm = ekat::ExeSpaceUtils<KT::ExeSpace>::view_reduction(team, 0, nlevs, qv_x_pdel);
+
+      EKAT_KERNEL_ASSERT_MSG(mm >= cc, "Error! Accumulated column moisture should be greater than surf_evap.\n");
+
+      Kokkos::parallel_for(Kokkos::TeamVectorRange(team, nlev_packs), [&](const int& k) {
+        const auto adjust = cc*qv_i(k)*pseudo_density_i(k)/mm;
+        qv_i(k) = (qv_i(k)*pseudo_density_i(k) - adjust)/pseudo_density_i(k);
+      });
+
+      surf_evap(i) = 0;
+    }
+  });
+}
```

Comment:
Questions about this function:
1. What is the significance of `surf_evap < 0`? I want to add a comment.
2. What does `surf_evap*dt*gravit` represent? Would like to use a more descriptive name.
3. Same as above for `qv*pseudo_density`.
4. For `mm>=cc` does my error message make sense? Is there something better to say?

Reply:
1) regarding surf_evap < 0 -- it means mass of vapor in the bottom layer will be reduced. i heard arguments that it is due to dew formation. you can say "surf_evap<0 may drive Q1 to negative values".

2) surf_evap units are kg/m^2/sec, so surf_evap*dt is change of mass in kg/m^2. since for tracers we often operate with quantities Q1*dp, which are in units of [dp]=kg/m/sec^2, the conversion between Pa and kg/m2 is 1/g factor. in other words, [surf_evap*dt*gravit] = Pa.

3) similarly, qv*pseudo_density is tracer 'mass' in Pa.

4) you can leave the message as is, or say ""Error! Total mass of column vapor should be greater than mass of QFLX.\n""